docs/modules/api_gateway.md

```diff
@@ -0,0 +1,92 @@
+# api_gateway
+
+**api_gateway** exposes a simple interface for specifying an API Gateway with the AWS [API V1 resources](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/api_gateway_rest_api). The module signature is patterned to be more declarative and similar to the [API Gateway V2 resources](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/apigatewayv2_api)
+
+Amazon exposes API gateways with 2 different APIs but they support different features. [Read More Here](https://www.tinystacks.com/blog-post/api-gateway-rest-vs-http-api-what-are-the-differences/) to see which version is right for your project.
+
+This module abstracts the V1 API since the V2 API is sufficiently well-designed that (for now) does not require an abstraction.
+
+
+## Example Usage
+
+---
+
+```terraform
+module "api_gateway" {
+  source = "github.com/RSS-Engineering/terraform.git?ref={commit}/modules/api_gateway"
+
+  name        = "API Gateway"
+  description = "REST Endpoints"
+  stage_name  = "v1"
+  tags        = {}
+
+  # Specify lambdas by (arbitrary) key and function-name for later reference via a route.
+  lambdas = {
+    "authorizer_lambda" : module.lambda_authorizer.function_name
+    "rest_api_handler" : module.lambda_rest_api.function_name
+  }
+  
+  # Send requests to / to the "rest_api_handler" lambda after passing through the authorizer
+  root_route = {
+    authorizer_key = "authorizer_lambda"
+    lambda_key     = "rest_api_handler"
+  }
+
+  routes = {
+    # Send GET requests to /categories to the proxy_url (bypassing the authorizer)
```

Comment:
Maybe have the route "keys" look like actual url paths: i.e. `/categories/`?  Also is this smart enough to handle nested routes: i.e. /categories/fruit

Reply:
For now, I'm considering nested routes as too complicated for this wrapper. But I've added the ability to prefix your routes with a slash for cosmetic reasons.